Algorithms.Tests/Search/FastSearcherTests.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System;
+using System.Linq;
+using Algorithms.Search;
+
+using NUnit.Framework;
+using NUnit.Framework.Internal;
+
+namespace Algorithms.Tests.Search
+{
+    public static class FastSearcherTests
+    {
+
+        [Test]
+        public static void FindIndex_ItemPresent_IndexCorrect()
+        {
+            // Arrange
+            var searcher = new FastSearcher();
+            var arr = new int[10];
+            for (var i = 0; i < arr.Length; i++)
+            {
+                arr[i] = i;
+            }
+
+            Array.Sort(arr);
+
+            for (int i = 0; i < arr.Length; i++)
+            {
+                int expectedIndex = i;
+                var x = arr[expectedIndex];
+                var actualIndex = searcher.FindIndex(arr, x);
+
+                // Assert
+                Assert.AreEqual(expectedIndex, actualIndex);
+            }
+        }
+
+        [Test]
+        public static void FindIndex_ItemPresent_IndexCorrect_ArrayIsOdd()
+        {
+            // Arrange
+            var searcher = new FastSearcher();
+            var arr = new int[11];
+            for (var i = 0; i < arr.Length; i++)
+            {
+                arr[i] = i;
+            }
+
+            Array.Sort(arr);
```

Comment:
What for? Isn't it sorted already?

Reply:
Of course it is, but one of the dependencies of the program is that the array must be sorted and I thought it would be easier to see it that way. I will remove it because it's redundant.